src/routes/rooms.ts

```diff
@@ -39,46 +36,28 @@ router.get(
   }),
 );
 
-// TODO: discuss if it's fine that types are verbs.
-type CreateRoom = {
-  name?: string;
-  description?: string;
-  options: Array<{
-    label?: string;
-    value?: string;
-  }>;
-};
-
 router.post(
   '/',
   asyncHandler(async (req, res) => {
-    const body: CreateRoom = req.body;
+    const newRoom: RoomInterface = req.body;
+    const { status, error, message } = validateRoom(newRoom);
 
-    if (!body) {
-      throw new UnprocessableEntityError(
-        ERROR_ROOM_CREATE_EMPTY_DATA,
-        'Empty body sent',
-      );
+    if (status == ValidationRoomStatus.INVALID) {
+      throw new UnprocessableEntityError(error, message);
     }
 
-    const hasOptions = body.options && Array.from(body.options).length > 0;
-    if (!hasOptions) {
-      throw new UnprocessableEntityError(
-        ERROR_ROOM_CREATE_EMPTY_OPTIONS,
-        'Options cannot be empty',
-      );
-    }
+    const room: Room = new Room();
+    room.name = newRoom.name;
+    room.description = newRoom.description || 'empty description';
 
-    const room = new Room();
-    room.name = body.name;
-    room.description = body.description;
-    room.options = body.options.map(params => {
-      const option = new Option();
+    room.options = newRoom.options.map(params => {
+      const option: OptionInterface = new Option();
```

Comment:
Here we have no problem with the ids because of the manual assignment

Reply:
But maybe we should also add the type just in case at some point we rely on having that value set (unlikely, but who knows)

We can achieve that by using `Omit`

Reply:
I'm not sure which one is better since typeorm is something new to me.  

But those interfaces are like compile-time safe checks, typeorm shouldn't be a problem if we are not even using the id when the model gets initialized. (maybe I'm missing something though)